.clang-tidy

```diff
@@ -0,0 +1,343 @@
+---
+FormatStyle: "file"
+WarningsAsErrors: "*"
+HeaderFilterRegex: ".*"
+Checks: "bugprone-argument-comment,
```

Comment:
This seems like a big list which we might not want all of. I think it would be better to slowly add ones over time.

I think we can default the `google-*` ones.

Reply:
I recommend [bugprone](https://clang.llvm.org/extra/clang-tidy/checks/list.htmll) checks first

Reply:
doesn't `bugprone-*` work? and then you can negate certain ones with `-bugprone-flying-aliens`?